scripts/export_comments.py

```diff
@@ -0,0 +1,103 @@
+import sys
+sys.path.append('..')
+import os
+import time
+
+from openpyxl import Workbook
+from sqlalchemy import create_engine
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.types import Boolean
+from sqlalchemy.dialects.postgresql import JSONB, UUID, TIMESTAMP
+
+import settings
+
+try:
+    db = create_engine(settings.DB_URI)
+    Session = sessionmaker(db)
+    session = Session()
+    base = declarative_base()
+except SQLAlchemyError as e:
+    print(e)
+    raise
+
+
+class SurveyResponse(base):
+    __tablename__ = 'responses'
+    tx_id = Column("tx_id",
+                   UUID,
+                   primary_key=True)
+
+    ts = Column("ts",
+                TIMESTAMP(timezone=True))
+
+    invalid = Column("invalid",
+                     Boolean,
+                     default=False)
+
+    data = Column("data", JSONB)
+
+    def __init__(self, tx_id, invalid, data):
+        self.tx_id = tx_id
+        self.invalid = invalid
+        self.data = data
+
+    def __repr__(self):
+        return '<SurveyResponse {}>'.format(self.tx_id)
+
+    def to_dict(self):
+        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
+
+
+def create_comments_book(survey_id, comments):
+    print("Generating Excel file")
+    wb = Workbook()
+    row = 2
+    ws = wb.active
+    ws.cell(1, 1, "Survey ID: {}".format(survey_id))
+    ws.cell(1, 2, "Comments found: {}".format(str(len(comments))))
+    for comment in comments:
+        row += 1
+        ws.cell(row, 1, comment.data['data']['146'])
+        ws.cell(row, 2, comment.data['submitted_at'])
+
+    filename = os.path.join(os.getcwd(), "comments_{}.xlsx".format(
+        time.strftime("%Y%m%d-%H%M%S")))
+    wb.save(filename)
+    wb.close()
+    print("Excel file generated")
+    return
+
+
+def get_all_comments_by_survey_id(survey_id, period):
+    # collect survey based on survey id and and 146 code
+    survey_records = session.query(SurveyResponse).filter(
+        SurveyResponse.data['survey_id'].astext == survey_id)
+    records = survey_records.filter(
+        SurveyResponse.data['collection']['period'].astext == period).all()
+    print("Retrieved {} comments".format(len(records)))
+    return records
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        sys.exit("Either survey id or period is missing")
+
+    survey_id = sys.argv[1]
+    period = sys.argv[2]
+
+    print('Survey id is {}'.format(survey_id))
+    print('Period is {}'.format(period))
+
+    try:
+        comments = get_all_comments_by_survey_id(survey_id, period)
+    except SQLAlchemyError as e:
+        print(e)
+        raise
+
+    if not comments or len(comments) == 0:
+        sys.exit("No comments, exiting script")
```

Comment:
I don't see the need to call `sys.exit`. The script will end here anyway.

Reply:
Not necessarily.  If there are no comments then it'll go ahead and create a file anyway that has nothing in it.  Might as well save the effort of cleanup by stopping it here.